connector-sdk/jackson-datatype-document/src/main/java/io/camunda/connector/document/annotation/jackson/DocumentReferenceModel.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.document.annotation.jackson;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.As;
+import io.camunda.connector.api.document.DocumentMetadata;
+import io.camunda.connector.api.document.DocumentReference;
+import io.camunda.connector.api.document.operation.DocumentOperation;
+import io.camunda.connector.document.annotation.jackson.DocumentReferenceModel.CamundaDocumentReferenceModel;
+import io.camunda.connector.document.annotation.jackson.DocumentReferenceModel.ExternalDocumentReferenceModel;
+import java.util.Map;
+import java.util.Optional;
+
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    property = DocumentReferenceModel.DISCRIMINATOR_KEY,
+    visible = true,
+    include = As.EXISTING_PROPERTY)
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = CamundaDocumentReferenceModel.class, name = "camunda"),
+  @JsonSubTypes.Type(value = ExternalDocumentReferenceModel.class, name = "external")
+})
+@JsonIgnoreProperties(ignoreUnknown = true)
+public sealed interface DocumentReferenceModel extends DocumentReference {
+
+  String DISCRIMINATOR_KEY = "$documentType";
```

Comment:
```suggestion
  String DISCRIMINATOR_KEY = "documentType";
```

Should we remove the $-sign as done here: https://github.com/camunda/camunda/pull/21080/files#diff-a18635d76659a781a6e8d301bd0fed7c77fd6cc5c8cfc57e11c5bbc19c78c9cdR1783 ?